### ISSUE TITLE
NP_NULL_ON_SOME_PATH_FROM_RETURN_VALUE false positive on org.springframework.web.multipart.MultipartFile::getOriginalFilename

### ISSUE THREAD

**Issue by ferng** [spotbugs/spotbugs#3789]
Sample code:
```java
    String originalFilename = "fileToProcess";
    if (multipartFile != null && multipartFile.getOriginalFilename() != null) {
      originalFilename = multipartFile.getOriginalFilename();
    }
```

throws the error inside the if, but it can't reach that line if anything is null

**Comment by gtoison** [comment]
Hello, SpotBugs does not know that multipartFile.getOriginalFilename() will return the same value on the second call.
Can you try extracting this into a local variable?

**Comment by Morph21** [comment]
This is such a weird take for this and overly conservative. 

In most cases this warning is a false positive. 

In theory it's correct that "a method" could return different results each call but for MultipartFile.getOriginalFilename() from spring the method itself is deterministic, it does not change between calls, it is not affected by concurrency in normal usage. You can't even set that filename so you would need to change variable multipartFile to get null inside if statement.

In my opinion this false positive is caused only because of "@Nullable" annotation that is on getOriginalFilename, nothing else.

Same issue is with ResponseEntity.getBody which also have @Nullable but realistically you would not change the body since you just check it after a call from API in most cases. Or you build it yourself to return to caller.
